modules/api/pkg/kubernetes/helper.go

```diff
@@ -37,18 +42,45 @@ func CheckContainerRuntime(ctx context.Context,
 	externalCluster *kubermaticv1.ExternalCluster,
 	externalClusterProvider provider.ExternalClusterProvider,
 ) (string, error) {
-	nodes, err := externalClusterProvider.ListNodes(ctx, masterClient, externalCluster)
+	clusterClient, err := externalClusterProvider.GetClient(ctx, masterClient, externalCluster)
+	if err != nil {
+		return "", err
+	}
+
+	controlPlaneNode, err := ListControlPlaneNode(ctx, clusterClient)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch container runtime: not able to list control plane nodes %w", err)
+	}
+
+	for _, node := range controlPlaneNode.Items {
+		containerRuntimeVersion := node.Status.NodeInfo.ContainerRuntimeVersion
+		strSlice := strings.Split(containerRuntimeVersion, ":")
+		return strSlice[0], nil
+	}
+
+	return "", fmt.Errorf("failed to fetch container runtime")
+}
+
+func ListControlPlaneNode(ctx context.Context, clusterClient ctrlruntimeclient.Client) (*corev1.NodeList, error) {
+	nodeReq, err := labels.NewRequirement(NodeControlPlaneLabel, selection.Exists, []string{})
 	if err != nil {
-		return "", fmt.Errorf("failed to fetch container runtime: not able to list nodes %w", err)
+		return nil, errors.Wrap(err, "error creating selector requirement")
 	}
-	for _, node := range nodes.Items {
-		if _, ok := node.Labels[NodeControlPlaneLabel]; ok {
-			containerRuntimeVersion := node.Status.NodeInfo.ContainerRuntimeVersion
-			strSlice := strings.Split(containerRuntimeVersion, ":")
-			for _, containerRuntime := range strSlice {
-				return containerRuntime, nil
-			}
-		}
+
+	selector := labels.NewSelector().Add(*nodeReq)
+	if err != nil {
+		return nil, errors.Wrap(err, "error converting node label requirement to selector")
 	}
-	return "", fmt.Errorf("failed to fetch container runtime: no control plane nodes found with label %s", NodeControlPlaneLabel)
+
+	nodes := &corev1.NodeList{}
+	listOpts := &ctrlruntimeclient.ListOptions{
+		Limit:         1,
```

Comment:
IMO, this should be a function parameter `limit` instead of constant (1), because the function signature doesn't make it clear that the function will return only one node for the list.